tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -0,0 +1,322 @@
+
+from streaming import MDSWriter
+from streaming.base.converters import dataframeToMDS
+
+MY_PREFIX = 'train'
+MY_BUCKET = 'mosaicml-composer-tests'
+LOCAL_MANUAL_TEST = False
+os.environ[
+    'OBJC_DISABLE_INITIALIZE_FORK_SAFETY'] = 'YES'  # set to yes to all fork process in spark calls
+
+
+@pytest.fixture(scope='class', autouse=True)
+def remote_local_dir() -> Any:
+    """Creates a temporary directory and then deletes it when the calling function is done."""
+
+    os.environ[
+        'GOOGLE_APPLICATION_CREDENTIALS'] = '/Users/xiaohan.zhang/.mosaic/mosaicml-research-nonprod-027345ddbdfd.json'
+
+    def _method(cloud_prefix: str = 'gs://') -> Tuple[str, str]:
+        mock_local_dir = mkdtemp()
+        mock_remote_dir = os.path.join(cloud_prefix, MY_BUCKET, MY_PREFIX)
+        return mock_local_dir, mock_remote_dir
+
+    return _method
+
+
+class TestDataFrameToMDS:
+
+    @pytest.fixture
+    def decimal_dataframe(self):
+        spark = SparkSession.builder.getOrCreate()  # pyright: ignore
+
+        schema = StructType([
+            StructField('id', IntegerType(), nullable=False),
+            StructField('name', StringType(), nullable=False),
+            StructField('amount', DecimalType(10, 2), nullable=False)
+        ])
+
+        data = [(1, 'Alice', Decimal('123.45')), (2, 'Bob', Decimal('67.89')),
+                (3, 'Charlie', Decimal('987.65'))]
+        df = spark.createDataFrame(data, schema)
+
+        yield df
+
+    @pytest.fixture
+    def dataframe(self):
+        spark = SparkSession.builder.getOrCreate()  # pyright: ignore
+
+        data = [('36636', 'Finance', (3000, 'USA')), ('40288', 'Finance', (5000, 'IND')),
+                ('42114', 'Sales', (3900, 'USA')), ('39192', 'Marketing', (2500, 'CAN')),
+                ('34534', 'Sales', (6500, 'USA'))]
+        schema = StructType([
+            StructField('id', StringType(), True),
+            StructField('dept', StringType(), True),
+            StructField(
+                'properties',
+                StructType([
+                    StructField('salary', IntegerType(), True),
+                    StructField('location', StringType(), True)
+                ]))
+        ])
+
+        df = spark.createDataFrame(data=data, schema=schema).repartition(3)
+        yield df
+
+    @pytest.mark.parametrize('keep_local', [True, False])
+    @pytest.mark.parametrize('merge_index', [True, False])
+    def test_end_to_end_conversion_local_nocolumns(self, dataframe: Any, keep_local: bool,
+                                                   merge_index: bool):
+        out = mkdtemp()
+        mds_kwargs = {
+            'out': out,
+            'keep_local': keep_local,
+            'compression': 'zstd:7',
+            'hashes': ['sha1', 'xxh64'],
+            'size_limit': 1 << 26
+        }
+
+        with pytest.raises(ValueError):
+            _, _ = dataframeToMDS(dataframe.select(col('id'), col('dept'), col('properties')),
```

Comment:
What value error this would throw ? You can add a `match` argument as part of `pytest.raises` which validates the error message, for example, something like [this](https://github.com/mosaicml/streaming/blob/main/tests/test_streaming.py#L283)